### ISSUE TITLE
[BUG] No bound on size of npm cache 

### ISSUE THREAD

**Issue by davisjam** [npm/cli#7006]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

I am trying to install many (100Ks) of npm packages for a research project. In this context, my npm cache grows until I exceed my user quota on the local compute cluster.

I reviewed the documentation for npm cache [here](https://docs.npmjs.com/cli/v10/commands/npm-cache). It does not appear to support a maximum cache size.

It appears my options are:
- Disable the cache
- Occasionally empty the cache myself

It would be nice if there were a third option:
- Configure my system with `npm cache --max-size 10GB --max-file 10K`

Questions:

1. I realize this is a pretty niche case. Are there other use cases (embedded systems?) where this would be helpful?
2. Is there some option I've missed that would give me an automated cap on the cache size?

### Expected Behavior

N/A

### Steps To Reproduce

N/A

### Environment

N/A

**Comment by wraithgar** [comment]
No there is no current way to do this.  The npm cache as implemented isn't really a proper cache, in that it doesn't remove expired items from itself.  It also doesn't currently keep track of its current size, so implementing this today would require a LOT of disk operations.

This would be an immense undertaking to implement and represent a significant overhaul of the current cache library: https://github.com/npm/cacache.

Folks who want to implement this are welcome to suggest ways to do this but it is not currently on the roadmap of the npm cli.

**Comment by prom00** [comment]
I came across this item, while cleaning up our servers. We've seen a slow increase of disk space usage.
After a quick inventarisation, we came to the conclusion our npm cache had grown to 2.8GB

Running a simple ```npm cache verify``` reduced it back to 305MB.

This isn't a very big issue, as it only grows steadily. But long term while running an application and upgrading packages, this will keep growing.

I'm not sure whether TS has this issue directly from the start, or also after a couple months/updates etc.
